### PR TITLE
Update travis config for Ubuntu Trusty images.

### PR DESCRIPTION
**Pull request by cwant**
Update the Travis config for Avalon for the now-default Trusty images.

By the way, if you are wondering about the Bundler stuff (downgrade from 1.15 to 1.14), here's what happens if you don't do that:

https://travis-ci.org/ualbertalib/avalon/builds/264524602

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -14,11 +16,16 @@ before_install:
   - cp config/secrets.travis.yml config/secrets.yml
   - git submodule init
   - git submodule update --depth=10 -- red5
+  - rvm use 2.3.3@global
```
Comment (murny) [inline .travis.yml]: Any reason for this? You are specifying this using travis' syntax `rvm: -2.3.3` at the bottom of this file?

File: .travis.yml [hunk 2]
```diff
@@ -14,11 +16,16 @@ before_install:
...
+  # Bundler 1.15 has stricter dependency checking -- problematic for Hydra projects
+  - gem uninstall -aIx bundler
+  - gem install bundler -v 1.14.6
   - bundle config without development:production:staging
 before_script:
   - rm config/database.yml
   - cp config/database.travis.yml config/database.yml
```
Comment (murny) [inline .travis.yml]: Why you need a custom yml file just for travis?

File: .travis.yml [hunk 3]
```diff
@@ -14,11 +16,16 @@ before_install:
...
   - mysql -e 'create database avalon_test;'
```
Comment (murny) [inline .travis.yml]: Any reason not to just do this through rails/rake? `rake db:create` etc? Or better yet in most rails projects now days they have a binstub for setting up your application called `bin/setup`

File: .travis.yml [hunk 4]
```diff
@@ -14,11 +16,16 @@ before_install:
   - cp config/secrets.travis.yml config/secrets.yml
   - git submodule init
   - git submodule update --depth=10 -- red5
+  - rvm use 2.3.3@global
+  # Bundler 1.15 has stricter dependency checking -- problematic for Hydra projects
+  - gem uninstall -aIx bundler
+  - gem install bundler -v 1.14.6
   - bundle config without development:production:staging
```
Comment (murny) [inline .travis.yml]: Your doing the config file way plus passing them in as arguments? Choose one?

```$ bundle config without development:production:staging
   $ bundle install --with mysql --without development debug```

File: .travis.yml (hunk 1 above)
Comment (cwant) [inline .travis.yml]: This appears to be required to revert the bundler version in the next line. Here's what happens if you don't have it: https://travis-ci.org/ualbertalib/avalon/jobs/264540700

File: .travis.yml (hunk 4 above)
Comment (cwant) [inline .travis.yml]: I think you misread, or I'm confused. There is a "gem install bundle" not a "bundle install" (maybe I didn't understand this comment).

Comment (cwant) [inline .travis.yml]: Ah, I see now, the bundler_args thingy at the top of the file. Will look into.

**Comment by cwant** [comment]
@murny I believe I have addressed the issues you brought up (thanks). Feel free to take another look when you get a chance.

File: .travis.yml [hunk 5]
```diff
@@ -16,19 +16,18 @@ before_install:
   - cp config/secrets.travis.yml config/secrets.yml
```
Comment (murny) [inline .travis.yml]: Changes look good. Just one small nitpick here..

Could we combine the secrets.travis.yml with the secrets.example.yml into one secrets.example.yml? (aka put in dummy secrets keys)

Then you dont have any specific travis configs in your app

File: .travis.yml [hunk 6]
```diff
@@ -14,14 +16,18 @@ before_install:
...
 before_script:
-  - rm config/database.yml
-  - cp config/database.travis.yml config/database.yml
-  - mysql -e 'create database avalon_test;'
+  - bundle exec rake db:create
   - bundle exec rake jetty:unzip jetty:config jetty:start delayed_job:start
   - bundle exec rake db:migrate
   - bundle exec rake avalon:db_migrate
 language: ruby
+cache: bundler
```
Comment (murny) [inline .travis.yml]: good call on this :+1:

File: .travis.yml (hunk 5 above)
Comment (cwant) [inline .travis.yml]: Done in latest commit.

**Pull request merged**